plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -64,4 +81,40 @@ public void onFailure(Exception e) {
         }
     }
 
+    @VisibleForTesting
+    void deleteModelChunks(String modelId, DeleteResponse deleteResponse, ActionListener<DeleteResponse> actionListener) {
+        DeleteByQueryRequest deleteModelsRequest = new DeleteByQueryRequest(ML_MODEL_INDEX);
+        deleteModelsRequest.setQuery(new TermsQueryBuilder(MODEL_ID_FIELD, modelId));
+
+        client.execute(DeleteByQueryAction.INSTANCE, deleteModelsRequest, ActionListener.wrap(r -> {
+            if ((r.getBulkFailures() == null || r.getBulkFailures().size() == 0)
+                && (r.getSearchFailures() == null || r.getSearchFailures().size() == 0)) {
+                log.info("All model chunks are deleted for model {}", modelId);
+                if (deleteResponse != null) {
+                    // If model metaData not found and deleteResponse is null, do not return here.
+                    // ResourceNotFound is returned to notify that this model was deleted.
+                    // This is a walk around to avoid cleaning up model leftovers. Will revisit if necessary.
+                    actionListener.onResponse(deleteResponse);
+                }
+            } else {
+                logFailure(r, modelId, actionListener);
```

Comment:
Seems this is not logging failure, just return correct error message, how about changing to `returnFailure`?

Reply:
It actually logged the full failure responses including all errors in line 117. "returnFailure" seems more reasonable though.